chains/solana/contracts/programs/ccip-router/src/instructions/v1/token_admin_registry.rs

```diff
@@ -5,74 +5,120 @@ use solana_program::{address_lookup_table::state::AddressLookupTable, log::sol_l
+
+    set_pending_administrator(token_mint, token_admin_registry, token_admin_registry_admin)?;
+
+    Ok(())
+}
+
+fn set_pending_administrator(
+    token_mint: Pubkey,
+    token_admin_registry: &mut Account<'_, crate::TokenAdminRegistry>,
+    new_admin: Pubkey,
+) -> Result<()> {
+    require_neq!(
+        new_admin,
+        Pubkey::new_from_array([0; 32]),
+        CcipRouterError::InvalidTokenAdminRegistryInputsZeroAddress
+    );
+
+    // TODO: Revert if administrator != 0 AlreadyRegistered
+
     token_admin_registry.version = 1;
-    token_admin_registry.administrator = token_admin_registry_admin;
-    token_admin_registry.pending_administrator = Pubkey::new_from_array([0; 32]);
+    token_admin_registry.administrator = Pubkey::new_from_array([0; 32]);
+    token_admin_registry.pending_administrator = new_admin;
     token_admin_registry.lookup_table = Pubkey::new_from_array([0; 32]);
 
-    emit!(AdministratorRegistered {
-        token_mint: mint,
-        administrator: token_admin_registry_admin,
+    emit!(AdministratorTransferRequested {
+        token: token_mint,
+        current_admin: Pubkey::new_from_array([0; 32]),
+        new_admin,
     });
 
     Ok(())
 }
 
-pub fn register_token_admin_registry_via_owner(
-    ctx: Context<RegisterTokenAdminRegistryViaOwner>,
+// Transfers the administrator role for a token to a new address with a 2-step process
+pub fn transfer_admin_role_token_admin_registry(
+    ctx: Context<ModifyTokenAdminRegistry>,
+    new_admin: Pubkey,
 ) -> Result<()> {
     let token_mint = ctx.accounts.mint.key().to_owned();
-    let mint_authority = ctx.accounts.mint.mint_authority.to_owned();
+    let token_admin_registry = &mut ctx.accounts.token_admin_registry;
 
-    require!(
-        mint_authority.is_some(),
-        CcipRouterError::InvalidInputsPoolAccounts
-    );
+    token_admin_registry.pending_administrator = new_admin;
 
-    let administrator = mint_authority.unwrap();
+    emit!(AdministratorTransferRequested {
+        token: token_mint,
+        current_admin: token_admin_registry.administrator,
+        new_admin,
+    });
 
+    Ok(())
+}
+
+// Accepts the admin role for a token
+pub fn accept_admin_role_token_admin_registry(
+    ctx: Context<AcceptAdminRoleTokenAdminRegistry>,
+) -> Result<()> {
+    let token_mint = ctx.accounts.mint.key().to_owned();
     let token_admin_registry = &mut ctx.accounts.token_admin_registry;
-    token_admin_registry.version = 1;
-    token_admin_registry.administrator = administrator;
+    let new_admin = token_admin_registry.pending_administrator;
+
+    token_admin_registry.administrator = new_admin;
     token_admin_registry.pending_administrator = Pubkey::new_from_array([0; 32]);
-    token_admin_registry.lookup_table = Pubkey::new_from_array([0; 32]);
 
-    emit!(AdministratorRegistered {
-        token_mint,
-        administrator,
+    emit!(AdministratorTransferred {
+        token: token_mint,
+        new_admin,
```

Comment:
Should this event have an `old_admin` entry? I see that most other storage change events reflect the inbound and outbound values, but I'm not sure if it's convention or coincidence.

Reply:
I copied the same format as EVM events: https://github.com/smartcontractkit/chainlink/blob/16f1529856a575c8d2091a16033a8d0371408d96/contracts/src/v0.8/ccip/tokenAdminRegistry/TokenAdminRegistry.sol#L28